packages/oidf-client/src/agent/OIDFClient.ts

```diff
@@ -1,81 +1,70 @@
 
-    if (cryptoServiceCallback !== undefined && cryptoServiceCallback !== null) {
-      DefaultCallbacks.setCryptoServiceDefault(cryptoServiceCallback)
-      DefaultCallbacks.setFetchServiceDefault(new DefaultFetchJSImpl())
-      // Depends on the crypto and fetch services, thus it must be the last one to be set
-      DefaultCallbacks.setTrustChainServiceDefault(new DefaultTrustChainJSImpl())
-      this.oidfClient = new FederationClient()
-    }
+    this.fetchServiceCallback = fetchServiceCallback
+    this.cryptoServiceCallback = cryptoServiceCallback
   }
 
-  readonly methods: IOIDFClient = {
-    resolveTrustChain: this.resolveTrustChain.bind(this),
-    signJwt: this.signJwt.bind(this),
-    verifyJwt: this.verifyJwt.bind(this),
+  private defaultCryptoJSImpl(context: IRequiredContext): ICryptoService {
+    return {
+      verify: async (jwt: string, key: JWK): Promise<boolean> => {
+        const verification = await context.agent.jwtVerifyJwsSignature({jws: jwt, jwk: key})
+        return verification.error === undefined
+      }
+    }
   }
 
-  private async resolveTrustChain(args: ResolveTrustChainArgs, context: RequiredContext): Promise<ResolveTrustChainCallbackResult> {
-    const { entityIdentifier, trustAnchors } = args
-    this.checkAndSetDefaultCryptoService(context)
-    return await this.oidfClient?.resolveTrustChain(entityIdentifier, trustAnchors)
-  }
+  private defaultFetchJSImpl(context: IRequiredContext): IFetchService {
+    return {
+      async fetchStatement(endpoint: string): Promise<string> {
+        const requestInfo = new Request(endpoint, {
+          method: 'GET'
+        })
 
-  private checkAndSetDefaultCryptoService(context: RequiredContext) {
-    if (
-      context.agent.jwtVerifyJwsSignature !== undefined &&
-      context.agent.jwtVerifyJwsSignature !== null &&
-      (this.oidfClient === undefined || this.oidfClient === null)
-    ) {
-      try {
-        DefaultCallbacks.setCryptoServiceDefault({
-          verify: async (jwt: string, key: any): Promise<boolean> => {
-            const jwk: JWK = { ...key }
-            try {
-              return !(
-                await context.agent.jwtVerifyJwsSignature({
-                  jws: jwt,
-                  jwk,
-                })
-              ).error
-            } catch (e) {
-              console.error(`Error verifying the JWT: ${e.message}`)
-              return Promise.reject(e)
-            }
-          },
+        const response = await context.agent.resourceResolve({
+          input: requestInfo,
+          resourceType: 'application/entity-statement+jwt',
         })
-        DefaultCallbacks.setFetchServiceDefault(new DefaultFetchJSImpl())
-        DefaultCallbacks.setTrustChainServiceDefault(new DefaultTrustChainJSImpl())
-        this.oidfClient = new FederationClient()
-      } catch (error) {
-        throw Error(`Could not initialize the federation client: ${error.message}`)
+
+        return await response.text()
```

Comment:
What are we doing with the HTTP status >= 300?  response.text() may return an error message or a blank string.

Reply:
I added a check to throw if status is not 200.